safespace/src/components/AddMessage.js

```diff
@@ -8,16 +8,58 @@ class AddMessage extends React.Component {
         this.state = {
             message : ''
         }
+    }
 
+    duration = (e) => {
+        const targetTime = e.target[2].value;
+        const currentHours = new Date().getHours();
+        const currentMinutes = new Date().getMinutes();
+        const pieces = targetTime.split(':');
+        const targetHours = parseInt(pieces[0],10);
+        const targetMinutes = parseInt(pieces[1],10);
+        const durationHours = targetHours - currentHours;
+        const durationMinutes = targetMinutes - currentMinutes;
+        const duration = durationHours * 60 * 60 * 1000 + durationMinutes * 60 * 1000
+        console.log(currentHours,currentMinutes,targetHours,targetMinutes,'duration: ',duration);
+        return duration;
     }
 
-    render() {
 
+    msgMaker = (e) => {
+        e.preventDefault();
+        console.log(e.target[1].value, 'time in msgMaker');
```

Comment:
while you can access the inputs values on the submit target object, you will see it more commonly utilizing controlled inputs and state tracking the value, you may find this article interesting https://goshakkk.name/controlled-vs-uncontrolled-inputs-react/